pkg/decision/experiment_whitelist_service.go

```diff
@@ -47,13 +47,16 @@ func (s ExperimentWhitelistService) GetDecision(decisionContext ExperimentDecisi
 		return decision, nil
 	}
 
-	variation, ok := decisionContext.Experiment.Variations[variationKey]
-	if !ok {
-		decision.Reason = reasons.InvalidWhitelistVariationAssignment
-		return decision, nil
+	// TODO(Matt): Add a VariationsByKey map to the Experiment struct, and use it to look up Variation by key
+	for _, variation := range decisionContext.Experiment.Variations {
+		variation := variation
```

Comment:
Why is this reassignment necessary? Can't you just use the variation assigned by the loop iterator?

Reply:
The linter forces it https://github.com/kyoh86/scopelint

Reply:
Oh... @pawels-optimizely thoughts on this? I won't block the merge on this 🙂

Reply:
Appears to be a valid catch for potentially hard to track down bugs. I also just learned something about go range loops :) In this particular case I believe we were safe since we're immediately returning with the current `variation`, but might as well be in the habit of "pinning".

Reply:
@pawels-optimizely The assignment is not just needed for go routines, that is just a common place it can manifest itself. See the `copies` example in scopelint readme.